patch_rebaser/patch_rebaser.py

```diff
@@ -172,10 +173,15 @@ def main():
     repo.branch.create(branch, remote_branch, reset_if_exists=True)
 
     # Rebase
-    LOGGER.info("Rebasing %s to %s", branch, commit)
-    repo.branch.rebase_to_hash(branch, commit)
-    LOGGER.info("Rebasing %s to %s", branch, remote_branch)
-    repo.branch.rebase_to_hash(branch, remote_branch)
+    try:
+        LOGGER.info("Rebasing %s to %s", branch, commit)
+        repo.branch.rebase_to_hash(branch, commit)
+        LOGGER.info("Rebasing %s to %s", branch, remote_branch)
+        repo.branch.rebase_to_hash(branch, remote_branch)
+    except git_exceptions.RebaseException:
+        LOGGER.info("Could not rebase. Cleaning up.")
+        repo.branch.abort_rebase()
```

Comment:
More of a question, if we fail in the second rebase then we are in an inbetween state. I am wondering if we should just drop a pre-rebase tag and then roll back to that tag to set everything back to it's original state. Not sure this is a good idea yet.

In my experience it is the second rebase that tends to fail more often then the first one. It maybe worth while to just try this an collect some data to see if this rollback works or if we need different rollbacks depending on which one fails.

Reply:
The branch will always be reset before the rebase begins (see https://github.com/release-depot/patch_rebaser/blob/master/patch_rebaser/patch_rebaser.py#L172 , just before the rebase() statements) so I don't think that is necessary. I believe jpena indicated as well that DLRN starts with a hard reset on master so there shouldn't be any in-between state to worry about there either.

Reply:
Perfect!